src/text-buffer.coffee

```diff
@@ -743,8 +746,12 @@ class TextBuffer
     lineEndings[lastIndex] = lastLineEnding
 
     # Replace lines in oldRange with new lines
-    spliceArray(@lines, startRow, rowCount, lines)
-    spliceArray(@lineEndings, startRow, rowCount, lineEndings)
+    if @lines.length > 1 or @lines[0].length > 0
```

Comment:
Yeah, I guess just copying an array into another array and then GCing the original can take a while. Here's the before (610ms):

![before](https://cloud.githubusercontent.com/assets/326587/21242786/cc634308-c2ca-11e6-8bb9-f0460cfa094c.png)

And after (345ms): 

![after](https://cloud.githubusercontent.com/assets/326587/21242796/de898dee-c2ca-11e6-8616-204375ed98eb.png)